ui/src/app/views/concept-add-modal/component.spec.tsx

```diff
@@ -0,0 +1,134 @@
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+import {ConceptAddModal} from './component';
+import {ConceptSetsApi} from 'generated/fetch/api';
+import {WorkspaceAccessLevel} from 'generated';
+import {currentWorkspaceStore} from 'app/utils/navigation';
+import {WorkspacesServiceStub} from 'testing/stubs/workspace-service-stub';
+
+
+describe('ConceptAddModal', () => {
+  let props;
+  let conceptSetsApi: ConceptSetsApiStub;
+  const stubConcepts = ConceptStubVariables.STUB_CONCEPTS;
+  const selectedDomain = DomainCountStubVariables.STUB_DOMAIN_COUNTS[0];
+
+  const component = () => {
+    return mount(<ConceptAddModal {...props}/>);
+  };
+
+  beforeEach(() => {
+    props = {
+      onSave: () => {},
+      onClose: () => {},
+      selectedConcepts: stubConcepts,
+      selectedDomain: selectedDomain
+    };
+
+    conceptSetsApi = new ConceptSetsApiStub();
+    registerApiClient(ConceptSetsApi, conceptSetsApi);
+  });
+
+  it('gets concept set list filtered by domain selected', async () => {
+    const wrapper = component();
+    currentWorkspaceStore.next({
+      ...WorkspacesServiceStub.stubWorkspace(),
+      accessLevel: WorkspaceAccessLevel.OWNER,
+    });
+    const stubConceptsInDomain = stubConcepts.filter((c) => c.domainId === selectedDomain.name);
+    expect(wrapper.find('[data-test-id="add-concept-title"]').first().text())
+        .toBe('Add ' + stubConceptsInDomain.length + ' Concepts to '
+            + selectedDomain.name + ' Concept Set');
+
+  });
+
+  it('displays option to add to existing concept set if concept set in domain exists', async () => {
+    const wrapper = component();
+    currentWorkspaceStore.next({
+      ...WorkspacesServiceStub.stubWorkspace(),
+      accessLevel: WorkspaceAccessLevel.OWNER,
+    });
+    const stubSetsInDomain = conceptSetsApi.conceptSets
+        .filter(s => s.domain === selectedDomain.domain)
+        .map(s => s.name);
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper.find('[data-test-id="add-to-existing"]').exists()).toBeTruthy();
+    const foundSets = wrapper.find('[data-test-id="existing-set"]').map((s) => s.text());
+    expect(foundSets).toEqual(stubSetsInDomain);
+
+  });
+
+  it('disables option to add to existing if concept set does not exist', async () => {
+    props.selectedDomain = DomainCountStubVariables.STUB_DOMAIN_COUNTS[2];
+    const wrapper = component();
+    currentWorkspaceStore.next({
+      ...WorkspacesServiceStub.stubWorkspace(),
+      accessLevel: WorkspaceAccessLevel.OWNER,
+    });
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper.find('[data-test-id="add-to-existing"]').exists()).toBeFalsy();
+    expect(wrapper.find('[data-test-id="create-new-set"]').exists()).toBeTruthy();
+    expect(wrapper.find('[data-test-id="toggle-existing-set"]')
+        .first().prop('disabled')).toBe(true);
+
+  });
+
+  it('defaults to create new set if set in domain does not exist', async () => {
```

Comment:
I think you could fold this into the test above, without increasing the scope too much. And then get rid of this one.